File: src/main/java/org/example/page/task/mypage/Service.java
package org.example.page.task.mypage;

import org.example.core.page.application.AbstractService;
import org.example.core.page.cookie.ICookie;
import org.example.page.cookie.store.Role;

class Service extends AbstractService {
    private final Request request;

    Service(Request request, ICookie cookie) {
        super(cookie);
        this.request = request;
    }

    MyProfile getMyProfile() {
        if (cookie instanceof Role role) {
            request.getMyProfile(role.getId());
            return new MyProfile(role.getId(), role.getRole());
        }
        return new MyProfile("default", "default");
    }
}

File: dev-page/src/main/kotlin/org/example/page/task/mypage/Service.kt
package org.example.page.task.mypage

import org.example.core.page.application.AbstractService
import org.example.core.page.cookie.ICookie
import org.example.page.cookie.store.Role

internal class Service(private val request: Request, cookie: ICookie?) : AbstractService(cookie) {
    val myProfile: MyProfile
        get() {
            (cookie as? Role)?. run {
                request.getMyProfile(this.id)
                return MyProfile(this.id, this.role)
            }
            return MyProfile("default", "default")
        }
}
